api/views/purchase.py

```diff
@@ -617,6 +611,16 @@ def _parse_errors(self, e, row):
         return errors
 
 
+class PurchasesDeleteView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def delete(self, request):
```

Comment:
Déso, j'ai oublié de préciser : ça devrait être un POST et non pas un DELETE. 

En deux mots : les requêtes HTTP DELETE ne sont pas censées avoir un body, et plusieurs implémentations ignorent le payload d'une requête DELETE.

Pour plus de précision, RFC 9110 dit : 

> Although request message framing is independent of the method used, content received in a DELETE request has no generally defined semantics, cannot alter the meaning or target of the request, and might lead some implementations to reject the request and close the connection because of its potential as a request smuggling attack 

Dans notre cas, le payload de la requête altère bien le comportement de la requête (et on aura potentiellement des implications dans certaines serveurs, surtout après l'intégration).

De plus plusieurs clients ne peuvent pas envoyer des payloads avec des DELETE (comme certaines SDK d'Android par exemple). Vu qu'on compte l'API est ouverte et que potentiellement on ouvrira cet endpoint, vaut mieux passer en POST.